Docs: Add usage insights export feature (#30376)
grafana/grafana: docs/sources/enterprise/usage-insights/export-logs.md

+++
title = "Export logs of usage insights"
description = "Export logs of usage insights"
keywords = ["grafana", "export", "usage-insights", "enterprise"]
aliases = ["/docs/grafana/latest/enterprise/usage-insights/export-logs.md"]
weight = 500
+++

# Export logs of usage insights

> **Note:** Available in Grafana Enterprise v7.4+.

By exporting usage logs to Loki, you can directly query them and create dashboards of the information that matters to you most, such as dashboard errors, most active organizations, or your top-10 most-used queries.

## Usage insights logs

> Usage insights logs are JSON objects that represent certain user activities:

Reviewer: ```suggestion
Usage insights logs are JSON objects that represent certain user activities, such as:
```

| `username`\* | string | Name of the Grafana user that made the request. |
| `userId`\* | number | ID of the Grafana user that made the request. |

## Configuration

The following procedure assumes that you have installed Loki. Refer to [Install Loki](/docs/loki/latest/installation/).

To export your logs, enable the usage insights feature and [configure]({{< relref "../../administration/configuration.md" >}}) an export location in the configuration file:

```ini
[usage_insights.export]
# Enable the usage insights export feature
enabled = true
# Storage type
storage = loki

[usage_insights.export.storage.loki]
# Set the communication protocol to use with Loki (can be grpc or http)
type = grpc
# Set the address for writing logs to Loki (format must be host:port)
url = localhost:9095
# Defaults to true. If true, it establishes a secure connection to Loki
tls = true
```

## Visualize Loki usage insights in Grafana

Now that your logs are exported into Loki, you can build Grafana dashboards to understand your Grafana instance usage.
1. Add Loki as a data source. Refer to [Grafana fundamentals tutorial](/tutorials/grafana-fundamentals/#6).
1. Import one of the following dashboards:

>     * <TODO> Link

Reviewer 2: @jongyllen do you have one or two dashboards we can ship?

+++
title = "Export logs of usage insights"
description = "Export logs of usage insights"
keywords = ["grafana", "export", "usage-insights", "enterprise"]
aliases = ["/docs/grafana/latest/enterprise/usage-insights/export-logs.md"]
weight = 500
+++

# Export logs of usage insights

> **Note:** Available in Grafana Enterprise v7.4+.

> By exporting usage logs to Loki, you can directly query them and create dashboards of the information that matters to you most, such as dashboard errors, most active organizations, or your top-10 most-used queries. Currently, usage logs can only be exported to Loki.

Reviewer 3: ```suggestion
By exporting usage logs to Loki, you can directly query them and create dashboards of the information that matters to you most, such as dashboard errors, most active organizations, or your top-10 most-used queries from within Loki.
```

Reviewer 3: Can we rephrase this to remove the sentiment of insufficiency that comes with the term "only"? Please check that the technical accuracy remains sound wrt my suggestion. Disregard is I have changed the meaning.

Reviewer 2: Made a tweak to remove the Loki mention at the end, since it is mentioned at the beginning. Agree with the suggestion to remove "only" :)

## Usage insights logs

> Usage insights logs are JSON objects that represent various user activity:

Reviewer 3: ```suggestion
Usage insights logs are JSON objects that represent certain user activities:
```

Reviewer 3: Rephased from (in this case) an uncountable noun to a countable noun because the items that follow are countable events.

- A user opens a dashboard.
- A query is sent to a data source.

### Scope

> A log is created every time a user opens a dashboard or when a query is sent to a data source in the dashboard view. A query that is made through Explore does not generate a log.

Reviewer 3: ```suggestion
A log is created every time a user opens a dashboard or when a query is sent to a data source in the dashboard view. A query that is performed via Explore does not generate a log.
```

Reviewer 3: My gut is that _perform_ is the more "natural" verb in this context because it is about place of execution rather than the specifics of creating a query.

    * [Usage insights](/grafana/dashboards/13785)
    * [Usage insights datasource details](/grafana/dashboards/13786)
1. Play with usage insights to understand them:
    * In Explore, you can use the query `{datasource="gdev-loki",kind="usage_insights"}` to retrieve all logs related to your `gdev-loki` data source.
    * In a dashboard, you can build a table panel with the query `topk(10, sum by (error) (count_over_time({kind="usage_insights", datasource="gdev-prometheus"} | json | error != "" [$__interval])))` to display the 10 most common errors your users see using the `gdev-prometheus` data source.
    * In a dashboard, you can build a graph panel with the queries `sum by(host) (count_over_time({kind="usage_insights"} | json | eventName="data-request" | error != "" [$__interval]))` and `sum by(host) (count_over_time({kind="usage_insights"} | json | eventName="data-request" | error = "" [$__interval]))` to show the evolution of the data request count over time. Using `by (host)` allows you to have more information for each Grafana server you have if you have set up Grafana for [high availability](<{{< relref "../../administration/set-up-for-high-availability.md" >}}>). 

> 
> 
> 
> 
> 

Reviewer 3: ```suggestion
```
